src/MaterialDesignControls/ControlsMaterial3/MaterialEntry.cs

```diff
@@ -0,0 +1,453 @@
+                txtEntry.CursorPosition = string.IsNullOrEmpty(textInsideInput) ? 0 : textInsideInput.Length;
+
+                if (string.IsNullOrEmpty(textInsideInput))
+                {
+                    await TransitionToTitle();
+                }
+            }
+            else
+            {
+                Unfocused?.Invoke(this, e);
+                UnfocusedCommand?.Execute(null);
+
+                if (string.IsNullOrEmpty(Text))
+                {
+                    await TransitionToPlaceholder();
+                }
+            }
+        }
+
+        private void TxtEntry_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            var changedByTextTransform = Text != null && txtEntry.Text != null && Text.ToLower() == txtEntry.Text.ToLower();
+
+            this.Text = this.txtEntry.Text;
+
+            if (!changedByTextTransform)
+            {
+                this.TextChangedCommand?.Execute(null);
+                this.TextChanged?.Invoke(this, e);
+            }
+
+            ApplyTextTransform();
+        }
+
+        private void FocusNextElement(int currentTabIndex)
+        {
+            try
+            {
+                var tabIndexes = this.GetTabIndexesOnParentPage(out int count);
+
+                if (tabIndexes != null)
+                {
+                    var nextElement = this.FindNextElement(true, tabIndexes, ref currentTabIndex);
+                    if (nextElement != null)
+                    {
+                        if (nextElement is Plugin.MaterialDesignControls.Material3.Implementations.CustomEntry nextEntry && nextEntry.IsEnabled && !nextEntry.IsReadOnly)
+                        {
+                            nextEntry.Focus();
+                            string textInsideInput = nextEntry.Text;
+                            nextEntry.CursorPosition = string.IsNullOrEmpty(textInsideInput) ? 0 : textInsideInput.Length;
+                        }
+                        else if (nextElement is CustomEditor nextEditor && nextEditor.IsEnabled && !nextEditor.IsReadOnly)
+                        {
+                            nextEditor.Focus();
+                        }
+                        else
+                        {
+                            this.FocusNextElement(++currentTabIndex);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                LoggerHelper.Log(ex);
+            }
+        }
+
+        private void ApplyTextTransform()
+        {
+            if (TextTransform == TextTransforms.Default || txtEntry.Text == null)
+                return;
+            else if (TextTransform == TextTransforms.Lowercase)
+                txtEntry.Text = txtEntry.Text.ToLower();
+            else if (TextTransform == TextTransforms.Uppercase)
+                txtEntry.Text = txtEntry.Text.ToUpper();
+        }
+
+        private async Task TransitionToTitle()
```

Comment:
As I can see in the material specs, the placeholder should be centered vertically (in the middle of the control).